relay-server/src/services/processor/playstation.rs

```diff
@@ -2,42 +2,328 @@
+    };
+    let mut runtime_context = RuntimeContext::default();
+
+    if let Some(hardware_id) = prospero.hardware_id.clone() {
+        event.server_name = Annotated::new(hardware_id);
+    }
+
+    // Elevate userdata values with sentry. prefix into the sentry event.
+    // Set the release based on the "release" key in the userdata
+    if let Some(release) = prospero
+        .userdata
+        .get("sentry.release")
+        .or_else(|| prospero.userdata.get("release"))
+    {
+        event.release = Annotated::new(LenientString(release.clone().into_owned()));
+    }
+    if let Some(environment) = prospero.userdata.get("sentry.environment") {
+        event.environment = Annotated::new(environment.clone().into_owned());
+    }
+    if let Some(username) = prospero.userdata.get("sentry.user.username") {
+        event.user.get_or_insert_with(Default::default).username =
+            Annotated::new(LenientString(username.to_string()));
+    }
+    if let Some(email) = prospero.userdata.get("sentry.user.email") {
+        event.user.get_or_insert_with(Default::default).email = Annotated::new(email.to_string());
+    }
+
+    let tags = event.tags.value_mut().get_or_insert_with(Tags::default);
+    macro_rules! add_tag {
+        ($key:expr, $value:expr) => {
+            tags.push(Annotated::new(TagEntry(
+                Annotated::new($key),
+                Annotated::new($value),
+            )));
+        };
+    }
+
+    add_tag!("cpu_vendor".into(), "Sony".into());
+    add_tag!("os.name".into(), "PlayStation".into());
+
+    let platform = "PS5";
+    runtime_context.name = Annotated::new(platform.into());
+    device_context.model = Annotated::new(platform.into());
+
+    add_tag!("cpu_brand".into(), format!("{platform} CPU"));
+    add_tag!("runtime.name".into(), platform.into());
+
+    if let Some(system_version) = &prospero.sdk_version {
+        add_tag!("os".into(), format!("PlayStation {system_version}"));
+        add_tag!("runtime".into(), system_version.into());
+        add_tag!("runtime.version".into(), system_version.into());
+
+        os_context.version = Annotated::new(system_version.into());
+        runtime_context.version = Annotated::new(system_version.into());
+    }
+
+    for (k, v) in &prospero.userdata {
+        if let Some(context_field_pair) = k.strip_prefix("sentry.context.") {
+            // Handle custom context data of the form: sentry.context.<name>.<field>
+            if let Some((context_name, field_name)) = context_field_pair.split_once('.') {
+                if let Context::Other(map) =
+                    contexts.get_or_insert_with(context_name, || Context::Other(Object::new()))
+                {
+                    map.insert(field_name.into(), Annotated::new(v.to_string().into()));
+                }
+            }
+        } else {
+            let tag = k.strip_prefix("sentry.").unwrap_or(k);
+            add_tag!(tag.into(), v.to_string());
+        }
+    }
```

Comment:
I think long-term we should clean this up a bit, with maybe what was proposed elsewhere (just allowing a generic json to set arbitrary event values) or by typing out a custom context, maybe something like a `Playstation` context.

Reply:
it's what @vaind is proposing for https://github.com/getsentry/sentry-playstation